### ISSUE TITLE
Bulk upload a dynamic store

### ISSUE THREAD

**Issue by silviurosu** [danielberkompas/elasticsearch-elixir#64]
I need to bulk upload custom stream built on runtime. For example I need to reindex all restaurants from a specific channel.
There is no way at the moment to send params to the store when building the stream.
Can we add the option to send params to Bulk.upload to be sent to the store? 
Or to expose at least a put_bulk method in Elasticsearch module that receives a list of items and perform a bulk upload?

**Comment by danielberkompas** [comment]
Yeah, that's a good suggestion. Overall, I would like to make the API of the package rely less on stores and more on plain old functions.

**Comment by silviurosu** [comment]
Can you give me more hints on how you want to tackle this issue? I want to be sure before digging in the code.
Should I add another method put_bulk? 
Or should I try to inject the stream in existing methods?
Or in case you want to more away from streams. then maybe we should suffice having a put_bulk that receives a list of documents and it up to user how it loads those documents?

**Comment by danielberkompas** [comment]
Hi @silviurosu, I'm currently working on a partial rewrite of the library that will maintain most of the current API, but expose the lower-level functions for direct bulk upload, etc. It should also eliminate the need for a store. Timeframe is hard to estimate right now, but I am working on it most weeks.

**Comment by silviurosu** [comment]
Ok, thanks for the notice. So I will not work on this and wait for your changes.
Let me know if I can help with something.
